### DISCUSSION TITLE
Can I get rounded corners on Wayland?

### DISCUSSION THREAD

**Discussion by afranke** [wezterm/wezterm#2751] [Q&A] [answered]
I’m running GNOME on Wayland. My config uses `window_decorations = "RESIZE"` and I’d like to have rounded corners like most modern GNOME apps do. It seems wezterm does that automatically on macOS. I don’t know if it should do it for all Wayland users (probably not), but there should at least be an option for it. I didn’t file a feature request because I was unable to determine whether it could already be achieved.

**Comment by wez** [comment, marked as answer]
Rounded corners are generally a feature of the desktop environment / compositor.  Wayland is a bit weird in that a number of features around window decoration/styling are not required to be handled by the compositor, and in the case of Gnome/Mutter, are instead assumed to be implemented by GTK applications.  This leads to non-GTK applications (such as wezterm) looking like ugly ducklings in the Gnome environment.  I generally recommend using XWayland over wayland, as I think the user experience is better.

It is not a goal of wezterm to look like a GTK/GNOME application.

It is technically possible to do something to round off the corners of the undecorated wezterm window.  I have no immediate plans for that, and would prefer to defer thinking about it until some other more pressing performance improvements are completed.

**Reply by afranke** [reply]
Shall I file a feature request, for that eventual day when those more pressing improvements are completed?

**Reply by wez** [reply]
If you like!

**Reply by Archuser202** [reply]
can i get it in windows
